src/NServiceBus.FileBasedRouting/FileBasedRoutingFeature.cs

```diff
@@ -112,7 +122,17 @@ public UpdateRoutingTask(Action updateRoutingCallback, TimeSpan routeFileUpdateI
 
             protected override Task OnStart(IMessageSession session)
             {
-                updateTimer = new Timer(state => updateRoutingCallback(), null, routeFileUpdateInterval, routeFileUpdateInterval);
+                updateTimer = new Timer(state =>
+                {
+                    try
+                    {
+                        updateRoutingCallback();
+                    }
+                    catch (Exception)
```

Comment:
but the callback rethrow the exception while this catch swallows it. We could move the catch from line 104 in here instead, but we couldn't log the file location as it won't be accessible here. Happy with that approach too.